integrations/airflow/marquez_airflow/extractors/great_expectations_extractor.py

```diff
@@ -0,0 +1,341 @@
+    """
+    Internal class to represent actual expectation values, per table and optionally per column
+    """
+    facet_key: str = attr.ib()
+    value: Any = attr.ib()
+    column_id: Optional[str] = attr.ib(default=None)
+
+
+class GreatExpectationsExtractorImpl(BaseExtractor):
+    operator_class = GreatExpectationsOperator
+
+    def __init__(self, operator):
+        super().__init__(operator)
+        self.operator._extractor = self
+        self.result = None
+
+    def store_result(self, result):
+        self.result = result
+
+    def parse_result(self) -> Optional[Dataset]:
+        try:
+            run_result = self.result['run_results']
+            # Great expectations generate long, impenetrable name here
+            validation_result = next(iter(run_result.values()))['validation_result']
+
+            data_quality_facet = self.parse_data_quality_facet(validation_result)
+            if not data_quality_facet:
+                return None
+
+            batch_kwargs = get_from_nullable_chain(
+                validation_result,
+                ['meta', 'batch_kwargs']
+            )
+            # To match dataset name we need canonical datasource name
+            name = batch_kwargs.get('datasource', None)
+            path = batch_kwargs.get('path', None)
+            return Dataset(
+                source=Source(
+                    name=name,
+                    type='GREAT_EXPECTATIONS',
+                    connection_url=path
+                ),
+                name=name,
+                type=DatasetType.DB_TABLE,
+                custom_facets={
+                    'dataQuality': data_quality_facet
+                }
+            )
+
+        except ValueError:
+            pass
+        return None
+
+    def parse_data_quality_facet(self, validation_result: Dict) \
+            -> Optional[DataQualityDatasetFacet]:
+        facet_data = {
+            "columnMetrics": defaultdict(dict)
+        }
+
+        # try to get to actual expectations results
+        try:
+            expectations_results = validation_result['results']
+            for expectation in expectations_results:
+                for parser in _EXPECTATIONS_PARSERS:
+
+                    # accept possible duplication, should have no difference in results
+                    if parser.can_accept(expectation):
+                        result = parser.parse_expectation_result(expectation)
+                        facet_data[result.facet_key] = result.value
+                for parser in _COLUMN_EXPECTATIONS_PARSER:
+                    if parser.can_accept(expectation):
+                        result = parser.parse_expectation_result(expectation)
+                        facet_data['columnMetrics'][result.column_id][result.facet_key] \
+                            = result.value
+
+            for key in facet_data['columnMetrics'].keys():
+                facet_data['columnMetrics'][key] = ColumnMetric(**facet_data['columnMetrics'][key])
```

Comment:
If I read this right, we're only collecting the actual observed values for everything and sticking it in the `ColumnMetric` type, is that right? Should we be collecting the assertions and the results?

Reply:
I assumed that we want to implement the API presented in the google doc, and the expectations themselves are just a way to get this data. If we want to collect them, then sure, we can.

Reply:
I would add a greatexpectations custome facet for the assertions

Reply:
Let's do it in follow up PR then